x-pack/dev-tools/functional_tests/lib/run_es_with_xpack.js

```diff
@@ -5,46 +5,45 @@
  */
 
 import { resolve } from 'path';
-import { createTestCluster } from '../../../../src/test_utils/es/es_test_cluster';
 import { log } from './log';
+import { Cluster } from '@kbn/es';
+import { RELATIVE_ES_BIN } from './paths';
 
 import { setupUsers, DEFAULT_SUPERUSER_PASS } from './auth';
 
-export async function runEsWithXpack({ ftrConfig, useSAML = false, from }) {
-  const cluster = createTestCluster({
-    port: ftrConfig.get('servers.elasticsearch.port'),
+export async function runEsWithXpack({ procs, config }) {
+  const cluster = new Cluster();
+  const pkg = require('../.././../../package.json');
+  const basePath = resolve(__dirname, '../../../../.es');
+  const randomHash = Math.random().toString(36).substring(2);
+  const clusterName = `test-${randomHash}`;
+  const installOptions = {
+    version: pkg.version,
+    installPath: resolve(basePath, clusterName),
+    sourcePath: resolve(__dirname, '../../../../../elasticsearch'),
     password: DEFAULT_SUPERUSER_PASS,
-    license: 'trial',
-    from,
-    log,
-  });
-
-  const kibanaPort = ftrConfig.get('servers.kibana.port');
-  const idpPath = resolve(
-    __dirname,
-    '../../../test/saml_api_integration/fixtures/idp_metadata.xml'
-  );
```

Comment:
@tylersmalley How does this work today? Originally we had this
```
    fs.createReadStream(
      resolve(XPACK_KIBANA_ROOT, 'test/saml_api_integration/fixtures/idp_metadata.xml')
    ).pipe(
      fs.createWriteStream(resolve(esExtractPath, 'config', 'idp_metadata.xml'))
    );
```
to copy the idp_metadata.xml file into the config dir of the elasticsearch install path. How is the current code able to reference this path without a permissions error? Locally (with my PR), I see

```
java.security.AccessControlException: access denied ("java.io.FilePermission" 
"/Users/archana/Development/apps/elastic/versions/master/kibana/x-pack/
test/saml_api_integration/fixtures/idp_metadata.xml" "read")
```

Reply:
You just need to pass it as an absolute path, kbn/es will ensure any absolute path to a file is copied to the config directory before running. 

https://github.com/elastic/kibana/blob/master/x-pack/dev-tools/functional_tests/lib/run_es_with_xpack.js#L38

https://github.com/elastic/kibana/blob/master/packages/kbn-es/src/utils/extract_config_files.js